bids/variables/variables.py

```diff
@@ -448,12 +448,31 @@ def resample(self, sampling_rate, inplace=False, kind='linear'):
         self.index = self._build_entity_index(self.run_info, sampling_rate)
 
         x = np.arange(n)
-        num = int(np.ceil(n * sampling_rate / old_sr))
+        # In Index we trust!
+        num = len(self.index)
+        # _build_entity_index computation above does it per run,
+        # which possibly provides a more stable solution and yadadada
+        num_computed = int(np.ceil(n * sampling_rate / old_sr))
```

Comment:
Do we want to change this to `np.round()` a la https://github.com/bids-standard/pybids/issues/361#issuecomment-458343256? And do we want to include f936aedcf36f715914816597d692ca37f9297f55 as well?